rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+        description: Current table identifier to rename and new table identifier to rename to
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        200:
+          description: OK
+        401:
+          description: Unauthorized
+        404:
+          description:
+            Not Found
+            - NoSuchTableException, Table to rename does not exist
+            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+                NamespaceToRenameToDoesNotExist:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          description: Not Acceptable - Unsupported Operation
+        409:
+          description: Conflict - The target table identifier to rename to already exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              example:
+                $ref: '#/components/examples/TableAlreadyExistsError'
+
+components:
+  #######################################################
+  # Common Parameter Definitions Used In Several Routes #
+  #######################################################
+  parameters:
+    namespace:
+      name: namespace
+      in: path
+      required: true
+      description:
+        A namespace identifier as a single string.
+        Multipart namespace parts should be separated by the null byte, %00.
+      schema:
+        type: string
+      examples:
+        singlepart_namespace:
+          value: "accounting"
+        multipart_namespace:
+          value: "accounting%00tax"
+
+    table:
+      name: table
+      in: path
+      description: A table name
+      required: true
+      schema:
+        type: string
+      example: "sales"
+
+    parent:
+      name: parent
+      in: query
+      description:
+        Optional parent namespace under which to list namespaces.
+        If parent is a multipart namespace, the parts must be separated by the null byte, %00.
+        When empty, list top-level namespaces.
+      required: false
+      schema:
+        type: string
+      example: "accounting%00tax"
+
+  ##############################
+  # Application Schema Objects #
+  ##############################
+  schemas:
+
+    ResponseDataObject:
+      type: object
+      description: JSON data payload returned in a successful response body
+      properties:
+        data:
+          type: object
+          description: Wrapper for the response of a successful request
+      example: { "data": { ... } }
+
+    ResponseErrorObject:
```

Comment:
Should these responses go under `responses`? Is there a distinction?

Reply:
There really isn't much of a distinction. It's possible to define a bit more things from within the `schema` section, but I'll check if I actually am using any of those anymore or not.

Reply:
So TIL it seems that originally there was just `schemas`. So it has some of the most support for JSON schema and things (like pattern checks that can be added as javax annotations etc).

Other things are newer and less feature rich. I've seen several tutorials with basic error classes, and they've mostly done it via `schemas`. I'm not sure if that's just to lessen the overhead for the article, but since we're getting rid of the `data` wrapper, I'm going to keep the `error` wrapper in schemas for now.

Keeping the `ResponseErrorObject` in schemas allows it to have much more control over what is and is not excepted, what's required to be defined, etc.

I'm open to moving it, but for now it seems like just getting rid of the `data` wrapper and the top level wrapper and keeping `ResponseErrorObject` would be good.